workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -476,5 +448,89 @@ private static IEnumerable<string> FindDirInPackages(string directory)
 
             return packagePaths.Union(cachedPackagePaths).Distinct();
         }
+
+        internal static void GenerateCodegenRunConfigs()
+        {
+            var toolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
+
+            var schemaCompilerPath = GetSchemaCompilerPath();
+            var logfilePath = toolsConfig.DefaultCodegenLogPath;
+
+            var codegenArgs = new List<string>
+            {
+                $"--json-dir=\"{ImprobableJsonDir}\"",
+                $"--schema-compiler-path=\"{schemaCompilerPath}\"",
+                $"--worker-json-dir=\"{WorkerJsonPath}\"",
+                $"--log-file=\"{logfilePath}\"",
+                $"--descriptor-dir=\"{toolsConfig.DescriptorOutputDir}\"",
+                $"--native-output-dir=\"{toolsConfig.FullCodegenOutputPath}\""
+            };
+
+            if (toolsConfig.VerboseLogging)
+            {
+                codegenArgs.Add("--verbose");
+            }
+
+            // Add user defined schema directories, warn if directory does not exist
+            foreach (var schemaDir in toolsConfig.SchemaSourceDirs)
```

Comment:
Could this be replaced with a `toolsConfig.Validate` or similar?
At this point the config should be valid.

Reply:
good catch - should enforce the tools config is valid at this point